course/utils.py

```diff
@@ -659,10 +664,28 @@ def has_permission(self, perm, argument=None):
         else:
             return (perm, argument) in self.permissions()
 
+    def _set_course_lang(self, action):
+        # type: (Text) -> None
+        if self.course.force_lang and self.course.force_lang.strip():
+                if action == "activate":
+                    self.old_language = translation.get_language()
+                    translation.activate(self.course.force_lang)
+                else:
+                    if self.old_language is not None:
+                        translation.activate(self.old_language)
+
     def __enter__(self):
+        if self._is_in_context_manager:
+            raise RuntimeError(
+                "Nested use of 'course_view' as context manager "
+                "is not allowed.")
+        self._is_in_context_manager = True
+        self._set_course_lang(action="activate")
```

Comment:
What about this? @inducer

Reply:
Or do you mean I need to add test on that?

Reply:
No, this is good.